lib/rust/parser/generate-java/java/org/enso/syntax2/Parser.java

```diff
@@ -21,28 +22,28 @@ private Parser(long stateIn) {
     private static native void freeState(long state);
     private static native ByteBuffer parseInput(long state, ByteBuffer input);
     private static native long getLastInputBase(long state);
+    private static native long getMetadata(long state);
+    static native long getUuidHigh(long metadata, long codeOffset, long codeLength);
+    static native long getUuidLow(long metadata, long codeOffset, long codeLength);
```

Comment:
I've written it once in some other PR, but: this will cross the JVM/native boundary frequently. Crossing the boundary is slow operation. _I also remember I got my answer - that it will be fixed in the future_.